File: source/downloads/code/LeetCode/Python/Combination-Sum4.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if not nums:
            return 0

        dp = [0]*(target+1)
        dp[0] = 1

        for i in range(target+1):
            for j in nums:
                if i+j <= target:
                    dp[i+j] += dp[i]
        return dp[target]

File: downloads/code/LeetCode/Java/Combination-Sum4.java
public class Solution {
    public int combinationSum4(int[] nums, int target) {
        int n = nums.length;
        if (n==0) return 0;
        int[] dp = new int[target+1];
        Arrays.sort(nums);
        for(int i=0; i<dp.length;i++){
            for(int num : nums){
                if(num > i)
                    break;
                if(num == i)
                    dp[i]++;
                else if(num < i)
                    dp[i] += dp[i-num];
                
            }
        
        }
        return dp[target];
    }
}
